### ISSUE TITLE
[css-shapes-2] Order of points and control points in `shape()`

### ISSUE THREAD

**Issue by smfr** [w3c/csswg-drafts#10666]
In https://github.com/w3c/csswg-drafts/issues/5841 we resolved that the points and the control points could be specified in any order, e.g. these are equivalent:
```
curve using 10px 30px by 20px 20px
curve by 20px 20px using 10px 30px
```

I'm having second thoughts. The `by` or `to` affect the interpretation of the control point coordinate pairs; they are either absolute (relative to the reference box), or relative to (the current point). So it's odd to specify the control points before you've given the `by` or `to`.

I think we should revert this part of the change.

**Comment by smfr** [comment]
@noamr

**Comment by noamr** [comment]
@smfr I think I've reached  a similar but slightly different conclusion. 
I think it's important to have `by` or `to` before the control points, but also it makes more sense to have the control points before the final point.

What do you think of https://github.com/w3c/csswg-drafts/issues/10644#issuecomment-2261027738?
Basically it's ordered, but the control points come first, and you can swap from `by` to `to` at any point. 
```
clip-path: shape(from top left,
  hline by 100px,
  curve by 100px 50px / to right bottom,
  curve by 50px 50px / 10% 20% / var(--x) var(--y),
  curve by 50px 50px / 10% 20% / to bottom 50%,
  smooth to bottom left / 0 0,
)
```

We can play with the exact syntax/separators.
This also allows using `position` in control points without losing the flexibility of `by`.

**Comment by dholbert** [comment]
@smfr, question for you, just to be explicit about where things stand:

The resolution here was "Make the order strict: endpoints before control points", but if I understood the discussion in https://github.com/w3c/csswg-drafts/issues/10649#issuecomment-2474354244 , we're relaxing that now and not requiring that strict order anymore. Is that correct?

**Comment by noamr** [comment]
> @smfr, question for you, just to be explicit about where things stand:
> 
> 
> 
> The resolution here was "Make the order strict: endpoints before control points", but if I understood the discussion in https://github.com/w3c/csswg-drafts/issues/10649#issuecomment-2474354244 , we're relaxing that now and not requiring that strict order anymore. Is that correct?

That's correct. Now that we have the `with` keyword we don't need the strict order.

**Comment by smfr** [comment]
I want to re-open this for further discussion. The allowed control point values depend on the command affinity (the to/by), so if the control points come first, you have to parse the control points in a relaxed way, and then go back and validate them after you hit the `to` or `by`. This doesn't feel great implementation-wise, but it also feels odd as an author to write out control points before you've written the `to` or `by` that determines how those control points are interpreted.

I thus propose that we go back to strict ordering of points and control points in `curve` commands.

**Comment by noamr** [comment]
> I want to re-open this for further discussion. The allowed control point values depend on the command affinity (the to/by), so if the control points come first, you have to parse the control points in a relaxed way, and then go back and validate them after you hit the `to` or `by`. This doesn't feel great implementation-wise, but it also feels odd as an author to write out control points before you've written the `to` or `by` that determines how those control points are interpreted.
> 
> I thus propose that we go back to strict ordering of points and control points in `curve` commands.

To be clear, you're proposing that the strict ordering would be to have the end point first, correct?
I'm OK with this

**Comment by smfr** [comment]
Yes, the `to` or `by` would have to come first. This would apply to `curve`, `smooth` and `arc` commands.
